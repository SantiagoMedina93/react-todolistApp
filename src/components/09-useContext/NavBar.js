import React from 'react';
import { Link, NavLink } from "react-router-dom";

/* DIFERENCIA ENTRE Link y NavLink: ESTILO => FOCALIZAR LA PAGINA EN LA QUE ESTA EL USUARIO*/


export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">

                <a className="navbar-brand" href="#">useContext</a>

                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <NavLink to="/" className={({ isActive }) => "nav-item nav-link" + (isActive ? " activated" : "")} >HOME</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink to="/about" className={({ isActive }) => "nav-item nav-link" + (isActive ? " activated" : "")}>ABOUT</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink to="/login" className={({ isActive }) => "nav-item nav-link" + (isActive ? " activated" : "")} >LOGIN</NavLink>
                        </li>
                        
                    </ul>
                </div>

            </div>
        </nav>
    )
};