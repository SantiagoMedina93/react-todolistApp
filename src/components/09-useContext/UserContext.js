import {createContext} from 'react';

//Higher order componenet => Tiene componentes dentro
export const UserContext = createContext(null);