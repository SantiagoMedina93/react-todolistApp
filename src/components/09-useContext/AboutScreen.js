import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const AboutScreen = () => {


    // CAMBIO EN EL CONTEXT = CAMBIO EN TODOS SUS HIJOS
    const {user, setUser}  = useContext(UserContext);
    
    const handleClick = () => {
        setUser({});
    }


    return (
        <>
            <h1>AboutScreen</h1>
            <hr />

            <pre>
                { JSON.stringify(user,null,3) }
            </pre>

            <button
                className='btn btn-warning'
                onClick={handleClick}
            >
                Logout
            </button>
        </>
    )
};