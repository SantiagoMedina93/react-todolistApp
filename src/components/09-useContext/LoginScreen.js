import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    // EL hook(useContext) buscara la informacion (variables, estados, fn)(izq) 
    // en el arbol de componentes uno que coincida con el definido como arg (der)
    const {setUser} = useContext(UserContext)
    
    const newUser = {
        id: 1234,
        name: 'Santiago'
    }

    
    return (
        <>
            <h1>LoginScreen</h1>
            <hr />

            <button
                className='btn btn-primary'
                onClick={ () => setUser(newUser) }
            >
                Login
            </button>
        </>
    )
};