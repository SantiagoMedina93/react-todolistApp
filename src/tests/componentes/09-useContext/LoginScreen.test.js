import React from "react";
import { mount } from "enzyme";

import { LoginScreen } from '../../../components/09-useContext/LoginScreen'
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Test in <LoginScreen.test />', () => {
    
    const setUser = jest.fn();
    
    const wrapper = mount( 
        <UserContext.Provider value={{
            setUser  // setUser = setUser:setUser
        }}>         
            <LoginScreen /> 
        </UserContext.Provider>
    )

    test('Debe de mostrarse corrcetamente', () => {

        expect( wrapper ).toMatchSnapshot();


    });

    test('Debe de ejecutar el setUser con el argumento esperado', () => {
        
        //SI hubiera usado el shallow solo se renderizaba el LoginScreen y no tengo el boton
        wrapper.find('button').prop('onClick')();

        expect( setUser ).toHaveBeenCalledWith({
            id:1234,
            name: 'Santiago'
        })
    

    });
});