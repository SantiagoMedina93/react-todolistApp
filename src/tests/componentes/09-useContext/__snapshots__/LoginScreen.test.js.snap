// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginScreen.test /> Debe de mostrarse corrcetamente 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
